### PR TITLE
Widgets: use Photon to display image in Image Widget.

### LINKED ISSUE #4141: Image Widget: Support Photon
If photon is enabled it would be great to use it for the image widget.

### PR DESCRIPTION
**Pull request by eliorivero**
Fixes #4141
#### Changes proposed in this Pull Request:
- image is loaded through Photon
#### Changelog

Extra Widgets: if Photon is available, use it to serve the image in the Image widget.

### CODE REVIEW

File: modules/widgets/image-widget.php [hunk 1]
```diff
@@ -70,7 +70,10 @@ public function widget( $args, $instance ) {
 
 		if ( '' != $instance['img_url'] ) {
 
-			$output = '<img src="' . esc_attr( $instance['img_url'] ) .'" ';
+			$output = '<img src="' . jetpack_photon_url( esc_attr( $instance['img_url'] ), array(
```
Comment (jeherve) [inline modules/widgets/image-widget.php]: Shouldn't it be `esc_url` instead of `esc_attr`?

Comment (bhubbard) [inline modules/widgets/image-widget.php]: @eliorivero Can you please update this pull request to use `esc_url()`

**Comment by jeherve** [comment]
I just tested it with an old image widget that was already active on my site, and for some reason `width` and `height` were set to 0.:

``

I can see `width` and `height` values in the widget settings, though.

screen shot 2016-10-27 at 21 27 58

No problem when taking the same image and creating a new Image widget, though:

``

I wonder if we should add a check like [here](https://github.com/Automattic/jetpack/blob/e868342b5967441ec933f19f7272e900f02a309a/modules/widgets/image-widget.php#L87) to cover old widgets like that?

**Comment by lancewillett** [comment]
Needs @eliorivero author reply before proceeding.

**Comment by kraftbj** [comment]
Do we want to always use Photon for this widget or only when the Photon module is enabled?

**Comment by lancewillett** [comment]
@kraftbj Based on the [original request](https://github.com/Automattic/jetpack/issues/4141), only when the module is enabled.

File: modules/widgets/image-widget.php [hunk 2]
```diff
@@ -70,7 +70,10 @@ public function widget( $args, $instance ) {
 
 		if ( '' != $instance['img_url'] ) {
 
-			$output = '<img src="' . esc_attr( $instance['img_url'] ) .'" ';
+			$output = '<img src="' . esc_url( jetpack_photon_url( $instance['img_url'], array(
```
Comment (kraftbj) [inline modules/widgets/image-widget.php]: Need to add `( method_exists( 'Jetpack', 'is_module_active' ) && Jetpack::is_module_active( 'photon' ) )` conditional.

**Comment by eliorivero** [comment]
@jeherve I can't reproduce the issue. Starting with master and Photon enabled, I added an Image widget an image URL and 400 x 225 like you have. Saved, checked in front end, all good. Switched to branch, checked in front end again, width and height were there.
That said, when you point this line `if ( '' != $instance['img_width'] ) {` I'm not really sure which condition you'd like to add to "cover old widgets". If you have something on your mind feel free to go ahead and add it.

File: modules/widgets/image-widget.php (hunk 2 above)
Comment (eliorivero) [inline modules/widgets/image-widget.php]: I added only `Jetpack::is_module_active( 'photon' )`. Note that in many widgets (like Follow Button) we call `Jetpack::is_active()` on `widgets_init` and only then the widget is registered so by the time of widget registration the `Jetpack` class (and its methods) exists.

**Comment by jeherve** [comment]
I'm afraid I can still experience an issue, but it now happens to new widgets, not existing ones. Try the following:

1. Check out your branch.
2. Go to Appearance > Widgets.
3. Drag a new Image Widget into a widget area.
4. Paste this URL in the URL field: `http://i.wpne.ws/igUH/Screen%20Shot%202017-01-09%20at%2011.26.00.png`
5. Hit the Save button at the bottom of the widget settings.
6. Load your site's front page.
7. The image is displayed, but no width or height parameters are added to the URL or to the image tag.

This doesn't happen with image widgets that were created before the branch was checked.

Compare this image widget, added before to check out the branch:

screen shot 2017-01-09 at 14 42 36

And this image widget, added after checking out the branch:

screen shot 2017-01-09 at 14 43 16

Can you reproduce as well?

**Comment by eliorivero** [comment]
I followed your steps and found that it only happens in my installation when Photon module is deactivated:

captura de pantalla 2017-02-06 a las 19 18 35

If it's active, the dimensions are properly set
captura de pantalla 2017-02-06 a las 19 19 30

**Pull request merged**